### DISCUSSION TITLE
How to get correctly the current address and how to log-out / discconnect the session?

### DISCUSSION THREAD

**Discussion by sebas6612** [WalletConnect/WalletConnectUnity#128] [Q&A] [answered]
Hi @skibitsky ,
I'm creating a connection using the Unity prefab, and is working, I can log in using Algorand blockchain and PeraWallet. but my question is how to get the current address and how to log out, I tried: 

1. Create a scene with a UI Button to connect the wallet.
2. Add the WalletConnectUnity prefab to the scene.
3. create a custom script with all the configurations to connect Algorand Mainnet then attach it to an object in the scene, and is working I can successfully connect from Pera wallet.
4. from another scene I want to get the current address and have a button to log-out / disconnect, I'm trying:

`[SerializeField] private WCSignClient _client;`
and assign the object previously configured in the Unity editor.
`_client.Disconnect();` but seems like is not working well
What I'm doing wrong? What is the correct way to do this? and also how to get the current account I am logged in with?

**Comment by skibitsky** [comment, marked as answer]
Hi @sebas6612,

Do you await `_client.Disconnect()`? `WCSignClient` is a singleton, have you tried accessing it with `WCSignClient.Instance` in your script?

It seems like you are using WalletConnectUnity v2. There is a demo app which shows how to implement Disconnect and how to get current address.

This code snippet returns the session, address, and chain ID:
```csharp
public (SessionStruct, string, string) GetCurrentAddress()             
{                                                                      
    var currentSession = _wcSignClient.Session.Get(_wc.Session.Keys[0]);         
                                                                       
    var defaultChain = currentSession.Namespaces.Keys.FirstOrDefault();
                                                                       
    if (string.IsNullOrWhiteSpace(defaultChain))                       
        return (default, null, null);                                  
                                                                       
    var defaultNamespace = currentSession.Namespaces[defaultChain];    
                                                                       
    if (defaultNamespace.Accounts.Length == 0)                         
        return (default, null, null);                                  
                                                                       
    var fullAddress = defaultNamespace.Accounts[0];                    
    var addressParts = fullAddress.Split(":");                         
                                                                       
    var address = addressParts[2];                                     
    var chainId = string.Join(':', addressParts.Take(2));              
                                                                       
    return (currentSession, address, chainId);                         
}                                                                      
```

In WalletConnectUnity v3.0.0 it can be simplified:
```csharp
var session = WalletConnect.Instance.ActiveSession;                                                               
var sessionNamespace = session.Namespaces;                                                                        
var address = WalletConnect.Instance.ActiveSession.CurrentAddress(sessionNamespace.Keys.FirstOrDefault()).Address;
```

**Reply by sebas6612** [reply]
Thanks for your help, I'll try to migrate to V3
